core/shared/platform/linux-sgx/sgx_thread.c

```diff
@@ -210,3 +210,58 @@ os_thread_get_stack_boundary()
     /* TODO: get sgx stack boundary */
     return NULL;
 }
+
+int
+os_rwlock_init(korp_rwlock *lock)
+{
+    assert(lock);
+
+    if (pthread_rwlock_init(lock, NULL) != BHT_OK)
+        return BHT_ERROR;
+
+    return BHT_OK;
```

Comment:
Had better use the macro `SGX_DISABLE_PTHREAD` to control the code, see other functions above:
```C
#ifndef SGX_DISABLE_PTHREAD
    assert(lock);

    if (pthread_rwlock_init(lock, NULL) != BHT_OK)
        return BHT_ERROR;
#endif

    return BHT_OK;
```

Reply:
updated